flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -877,6 +913,20 @@ public long getWriteBatchSize() {
 			WRITE_BATCH_SIZE.defaultValue().getBytes() : writeBatchSize;
 	}
 
+	/**
+	 * Gets the read buffer size for read checkpoint stream.
+	 *
+	 * <p>If not explicitly configured, this is the default value of
+	 * {@link CheckpointingOptions#FS_READ_BUFFER_SIZE}.
+	 *
+	 * @return The file system read buffer size, in bytes.
+	 */
+	public int getFsReadBufferSize() {
```

Comment:
As you can see, there existed three duplicate `getFsReadBufferSize` in your PR, which could be an abstract base method for all state backends. 
If we think it one step further, we now has same interfaces for checkpoint write of state backend, e.g. `createCheckpointStorage`, however, we do not have same interfaces for checkpoint **read** of state backend. That's why you meet such embarrassment here. 
I think this deserve a discussion over FLINK-19465 and FLIP-142, @sjwiesman what do you think of this?

Reply:
Thanks to @Myasuka 's perfect analysis, this is indeed the reason for my design.